cmake/modules/AddSwift.cmake

```diff
@@ -1833,7 +1840,21 @@ function(_add_swift_executable_single name)
       BINARY_DIR ${SWIFT_RUNTIME_OUTPUT_INTDIR}
       LIBRARY_DIR ${SWIFT_LIBRARY_OUTPUT_INTDIR})
 
+  # On Windows, libraries are prefixed with "lib", so we need to adjust FAT_LIBRARIES for this.
+  if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+    _list_add_string_prefix(
+        "${SWIFTEXE_SINGLE_LINK_FAT_LIBRARIES}"
+        "lib"
+        SWIFTEXE_SINGLE_LINK_FAT_LIBRARIES)
+  endif()
+
   target_link_libraries("${name}" ${SWIFTEXE_SINGLE_LINK_LIBRARIES} ${SWIFTEXE_SINGLE_LINK_FAT_LIBRARIES})
+
+  # On Windows, we need to link rpcrt4.lib for UUIDs and mincore.lib for various file system APIs
+  if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+    target_link_libraries("${name}" "mincore.lib" "rpcrt4.lib")
+  endif()
```

Comment:
I'm linking rpcrt4.lib for creation of UUIDs that doesn't rely on libuuid which is tricky to build on Windows.
I'm linking mincore.lib in order to fix some unresolved symbol errors when linking Clang libraries, as Clang uses file system APIs found in mincore